account_cutoff_accrual_picking/models/account_cutoff.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Account Cut-off Accrual Picking module for OpenERP
+#    Copyright (C) 2013 Akretion (http://www.akretion.com)
+#    @author Alexis de Lattre <alexis.delattre@akretion.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from odoo import _, api, exceptions, fields, models
+
+class AccountCutoff(models.Model):
+    _inherit = 'account.cutoff'
+
+    @api.model
+    def _inherit_default_cutoff_account_id(self):
+        """ Set up default account for a new cutoff """
+        account_id = super(AccountCutoff,
+                           self)._inherit_default_cutoff_account_id()
+        type_cutoff = self.env.context.get('type')
+        company = self.env.user.company_id
+        if type_cutoff == 'accrued_expense':
+            account_id = company.default_accrued_expense_account_id.id or False
+        elif type_cutoff == 'accrued_revenue':
+            account_id = company.default_accrued_revenue_account_id.id or False
+        return account_id
+
+    def _prepare_lines(self, line, account_mapping):
+        """
+        Calculate accrued expense using purchase.order.line
+        or accrued revenu using sale.order.line
+        """
+        assert self.type in ('accrued_expense', 'accrued_revenue'),\
+            "The field 'type' has a wrong value"
+        company_currency_id = self.company_id.currency_id
+        currency = line.currency_id
+        if self.type == 'accrued_expense':
+            # Processing purchase order line
+            account_id = line.product_id.property_account_expense_id.id
+            if not account_id:
+                account_id = line.product_id.product_tmpl_id.categ_id.\
+                    property_account_expense_categ_id.id
+            if not account_id:
+                raise exceptions.Warning(
+                    _("Error: Missing expense account on product '%s' or on "
+                        "related product category.") % (line.product_id.name))
+            analytic_account_id = line.account_analytic_id.id or False
```

Comment:
`or False` can be removed: if there is no analytic account, it will return an empty record which will be evaluated as `False` too